### PR TITLE
refactor: refactor app for maintainability

### PR DESCRIPTION
**Pull request by robinmuhia**
> Add github workflows for CI/CD, stale issues and dependabot configuration

> Abstract all apps into one folder for easier maintainability

> Add a common model to add common configurations such as base mixins, serializers etc

> Segregrate requirements into dev, test and prod requirements

> Segregrate test folders for easier testing

> Add formatting and lint checks

> Add entrypoint.sh to run migrations and start container

> Add Makefile

> Add pytest.ini configurations for testing

> Snake_case pollingStation to meet python PEP8 standards

> Remove docker-compose.yml configuration to allow for various deployment strategies

### CODE REVIEW

File: docker-compose.yml
Comment (C-o-m-o-n) [inline docker-compose.yml:1]: Hi @robinmuhia  I have staarted reviewing your pr. I noiced that you deleted this file. It is however needed to run the app. If there's a specific reason for removing it, kindly let me know, and let me know howI should run the app instead.

Most of all, KEEP THE GOOD WORK GOING!!
Happy hacking with you :)

Comment (robinmuhia) [reply docker-compose.yml:1]: yes , i deleted it as entrypoint.sh is what will run the application.

Comment (C-o-m-o-n) [reply docker-compose.yml:1]: The directive whas to use docker. I like the way you structured it though.  Can we find a way to still work with Docker?

Comment (robinmuhia) [reply docker-compose.yml:1]: It works with docker.

What OS are you using? 

It might be an issue with chmod permissions.

Comment (robinmuhia) [reply docker-compose.yml:1]: Quick one, could you please conclude this PR? Rebasing it is very time-consuming

**Pull request merged**